module/documents/activity/attack.mjs

```diff
@@ -84,6 +84,11 @@ export default class AttackActivity extends ActivityMixin(AttackActivityData) {
     const { parts, data } = this.getAttackData();
     const targets = getTargetDescriptors();
 
+    if ( (this.item.type === "weapon") && (this.item.system.quantity === 0) ) {
+      ui.notifications.error("DND5E.ATTACK.Warning.NoQuanity", { localize: true });
```

Comment:
We do prevent selecting a specific ammunition item if its quantity is zero, but you can always bypass that and select blank ammunition so maybe this could just a non-stopping warning.